coldfront/core/user/forms.py

```diff
@@ -95,25 +95,35 @@ def clean_email(self):
             raise forms.ValidationError(mark_safe(message))
         return email
 
-    def clean_middle_name(self):
-        cleaned_data = super().clean()
-        self.middle_name = cleaned_data.pop('middle_name', '')
-        return cleaned_data
-
     def clean_phone_number(self):
         cleaned_data = super().clean()
         self.phone_number = cleaned_data.pop('phone_number', '')
         return cleaned_data
 
+    def clean_first_name(self):
+        cleaned_data = super().clean()
```

Comment:
This is something that just comes with Django knowledge: the `clean_*` methods for individual fields are called _after_ the `clean` method has already been run, so there shouldn't be a need to run it within each of these methods.

You can retrieve the cleaned data from `self.cleaned_data`. (In fact, this is done in line 115 below.)

Here's the relevant [documentation](https://docs.djangoproject.com/en/3.2/ref/forms/validation/#form-and-field-validation). See the fourth and fifth bullet points.

Reply:
Also, the `pop` shouldn't be necessary. I think this should do the trick:

```
return self.cleaned_data['first_name'].title()
```

The return value of this method will be accessible from the form's `cleaned_data` back in the `UserProfileUpdate` view.